src/app/loginComponents/accounts/controls/delete-account-dialog/delete-account-dialog.component.ts

```diff
@@ -0,0 +1,98 @@
+import { Component } from '@angular/core';
+import { AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { MatDialogRef, MatSnackBar } from '@angular/material';
+import { map } from 'rxjs/operators';
+
+import { LogoutService } from '../../../../shared/logout.service';
+import { User } from '../../../../shared/swagger';
+import { UserService } from '../../../user.service';
+
+@Component({
+  selector: 'app-delete-account-dialog',
+  templateUrl: './delete-account-dialog.component.html',
+  styleUrls: ['./delete-account-dialog.component.scss']
+})
+export class DeleteAccountDialogComponent {
+  username = '';
+  usernameFormControl: FormControl;
+  usernameForm: FormGroup;
+
+  constructor(public userService: UserService, public form: FormBuilder, public dialogRef: MatDialogRef<DeleteAccountDialogComponent>,
+    private logoutService: LogoutService, private matSnackBar: MatSnackBar) {
+    this.userService.user$.pipe(map((user: User) => {
+      return user.username;
+    })).subscribe((username: string) => {
+      this.setupForm(username);
+    });
+  }
+
+  /**
+   * Sets up the form
+   *
+   * @param {string} username  The user's actual username
+   * @memberof DeleteAccountDialogComponent
+   */
+  setupForm(username: string): void {
+    this.username = username;
+    this.usernameFormControl = new FormControl('', [Validators.required, this.validateUsername(this.username)]);
+    this.usernameForm = new FormGroup({
+      usernameFormControl: this.usernameFormControl
+    });
+  }
+
+  // Close dialog
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  /**
+   * Delete the user account
+   *
+   * @memberof DeleteAccountDialogComponent
+   */
+  deleteAccount(): void {
+    this.deleteAccountSuccess();
+  }
+
+  /**
+   * What happens when deleting the account has succeeded.
+   *
+   * @private
+   * @memberof DeleteAccountDialogComponent
+   */
+  private deleteAccountSuccess(): void {
+    this.logoutService.logout();
+    this.matSnackBar.open('Deleting Dockstore account succeeded', 'Dismiss', {
+      duration: 5000,
+    });
+  }
+
+  /**
+   * What happens when deleteing the account has failed.
+   *
+   * @private
+   * @memberof DeleteAccountDialogComponent
+   */
+  private deleteAccountFailure(): void {
+    this.matSnackBar.open('Deleting Dockstore account failed', 'Dismiss', {
+      duration: 5000,
```

Comment:
We should probably have this 5000 value in a shared location; I copied the 5000 for permissions component as well. Not for this PR, just mentioning it.

Reply:
Should be straightforward, I've done something similar for tooltips already.